mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -183,24 +191,18 @@ private[ml] trait DecisionTreeParams extends PredictorParams
  */
 private[ml] trait TreeClassifierParams extends Params {
 
+  // Impurity should be overriden when setting a default. This should be a def, but has
```

Comment:
typo: overridden

Also, did you try generating the docs?  If you use a def here but override it elsewhere as a val, then does it show up as a val in the docs?